.golangci.yml

```diff
@@ -1,17 +1,30 @@
-linters-settings:
-  govet:
-    check-shadowing: true
-
 # TODO_TECHDEBT: Enable each linter listed, 1 by 1, fixing issues as they appear.
 # Don't forget to delete the `disable-all: true` line as well.
 linters:
   disable-all: true
   enable:
-#    - govet
-#    - revive
-#    - errcheck
-#    - unused
+    #    - govet
+    #    - revive
+    #    - errcheck
+    #    - unused
+    - gofumpt
     - goimports
+    - lll
+    - gci
+
+linters-settings:
+  govet:
+    check-shadowing: true
+  lll:
+    line-length: 120
+    tab-width: 4
+  gci:
+    sections:
+      - standard # std lib
+      - default # external
+      - prefix(github.com/pokt-network/poktroll) # local imports
+    skip-generated: true
```

Comment:
Can we remove this makefile target and script now?
- https://github.com/pokt-network/poktroll/blob/main/Makefile#L155
- https://github.com/pokt-network/poktroll/blob/main/tools/scripts/goimports/main.go

Reply:
Or perhaps it still makes sense to have the make target to be able to attempt to fix fixable issues. :thinking:

Reply:
> I believe so will have to make sure this works first but running `golangci-lint` should do the job of these two

The reason why it's not a simple as running `golangci-lint` is because there is some additional generated code that we need to ignore. The silliest example is the multi-line import statement which ignite seems to rely on but the linter would rewrite as a single line. :roll_eyes: This is the reason for the script package's existend.

Reply:
running `golangci-lint run` should do show errors and idk if it fixes for u this PR probs gunna have to include me fixing everything that doesn't fit the rules anyway

Reply:
> The silliest example is the files that contain a multi-line import statement that only has one package, which ignite seems to rely on, but the linter would rewrite as a single line

An example of one of these files would help me a lot @bryanchriswhite

Reply:
I have replaced the tool you mentioned and added another and same with makefiles that are relevant to the new rules